### DISCUSSION TITLE
Deta/Drive :  "Bad Gateway" error

### DISCUSSION THREAD

**Discussion by dlink1999** [deta/cloud-docs#213] [Help] [answered]
Hi, there:

I am pretty new in fastapi and deta.

I follow the tutorial code, and deploy micro successfully.
But when I try to initiate the drive, it doesn't work.

```py
from fastapi import FastAPI, File, UploadFile
from fastapi.responses import HTMLResponse, StreamingResponse
from deta import Deta

app = FastAPI()
deta = Deta() # configure your Deta project
drive = deta.Drive("photos") # access to your drive"
```

The Visor shows the following message. What do I miss?

```
"Traceback (most recent call last):
File "/opt/python/detalib/debugger.py", line 142, in wrap
result = func(event, context)
File "/var/task/_entry.py", line 12, in handler
import main # noqa
File "/var/task/main.py", line 7, in
drive = deta.Drive("photos") # access to your drive
AttributeError: 'Deta' object has no attribute 'Drive'"
```

**Comment by abdelhai** [comment]
hi @dlink1999 

add `deta` to your **requirements.txt** then run `deta deploy`

**Reply by dlink1999** [reply]
Thanks.

  The totorial suggests

fastapi
uvicorn
deta
python-multipart 

  as requirements.txt

  and remove "uvicorn" when usung default project.

  But the error result remains the same.

**Comment by abdelhai** [comment, marked as answer]
I just ran the code you provided as is with no errors. the error just says that you have an old deta sdk installed and by adding `deta` to the reqs file and re-deploying the problem should be resolved.

one thing you could try is to remove `deta` deploy, then add it back in again and then redeploy.

**Reply by dlink1999** [reply]
Really appreciated. I'll try again later on.
